server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStatsResponse.java

```diff
@@ -41,30 +50,85 @@ public RemoteStoreStatsResponse(
         List<DefaultShardOperationFailedException> shardFailures
     ) {
         super(totalShards, successfulShards, failedShards, shardFailures);
-        this.shards = shards;
+        this.remoteStoreStats = shards;
     }
 
-    public RemoteStoreStats[] getShards() {
-        return this.shards;
+    public RemoteStoreStats[] getRemoteStoreStats() {
+        return this.remoteStoreStats;
     }
 
     public RemoteStoreStats getAt(int position) {
-        return shards[position];
+        return remoteStoreStats[position];
+    }
+
+    public Map<String, Map<Integer, List<RemoteStoreStats>>> groupByIndexAndShards() {
+        if (indexWiseStats != null) {
+            return indexWiseStats;
+        }
+        Set<String> indexNames = new HashSet<>();
+        for (RemoteStoreStats shardStat : remoteStoreStats) {
+            String indexName = shardStat.getShardRouting().getIndexName();
+            indexNames.add(indexName);
+        }
+
+        Map<String, Map<Integer, List<RemoteStoreStats>>> indexWiseStats = new HashMap<>();
+        for (String indexName : indexNames) {
+            Set<RemoteStoreStats> perIndexStats = new HashSet<>();
+            for (RemoteStoreStats shardStat : remoteStoreStats) {
+                if (shardStat.getShardRouting().getIndexName().equals(indexName)) {
+                    perIndexStats.add(shardStat);
+                }
+            }
+            indexWiseStats.put(indexName, groupByShards(perIndexStats));
+        }
+        this.indexWiseStats = indexWiseStats;
+        return indexWiseStats;
```

Comment:
We can further optimise the code by using the `computeIfAbsent` method of the Map.

Reply:
Ack. Will check on this.